### PR TITLE
feat: add animator component

### PR DESCRIPTION
**Pull request by Suduck**
## What does this PR change?

This PR adds the animator component for the new ECS 7

## How to test the changes?

Can't be tested

## Our Code Review Standards

https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md

### CODE REVIEW

**Review by AjimenezDCL — APPROVED** [review]

File: unity-renderer/Assets/DCLPlugins/ECS7/ECSComponents/Animator/Handler/AnimatorComponentHandler.cs [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+            if (entityId != entity.entityId)
+                return;
+          
+            dataStore.animatorShapesReady.OnAdded -= AnimatorShapeReadyAdded;
+            
+            Initialize(entity);
+            isShapeLoaded = true;
+
+            UpdateAnimationState(model);
+        }
```
Comment (pbosio) [inline unity-renderer/Assets/DCLPlugins/ECS7/ECSComponents/Animator/Handler/AnimatorComponentHandler.cs:71]: what would happen if the gltfshape component is changed after adding the animator?
i.e:
```
add shape = gltf1
add animator
modify shape = glf2
```

File: unity-renderer/Assets/DCLPlugins/ECS7/ECSComponents/Animator/Handler/AnimatorComponentHandler.cs [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+                    unityState.speed = model.States[i].Speed;
+                    unityState.enabled = model.States[i].Playing;
+
+                    if (model.States[i].ShouldReset)
+                        ResetAnimation(model.States[i], unityState.clip);
+
+                    if (model.States[i].Playing && !animComponent.IsPlaying(model.States[i].Clip))
+                        animComponent.Play(model.States[i].Clip);
+                }
+            }
```
Comment (pbosio) [inline unity-renderer/Assets/DCLPlugins/ECS7/ECSComponents/Animator/Handler/AnimatorComponentHandler.cs:129]: maybe we could use this opportunity to do a proper refactor on this component? It was one of Brian's dreams.
cause I think this is the worst performing component we have. the code with, all those dictionaries that look almost redundant, is hard to follow and there are a lot of code for hotfixes there

File: unity-renderer/Assets/DCLPlugins/ECS7/ECSComponents/GLTFShape/Handler/GLTFShapeComponentHandler.cs [hunk 3]
```diff
@@ -8,17 +8,20 @@
...
         internal PBGLTFShape model;
         internal readonly LoadWrapper_GLTF loadWrapper;
+        internal IDCLEntity entity;
 
+        private readonly ShapeRepresentation shapeRepresentation;
         private readonly DataStore_ECS7 dataStore;
 
         public GLTFShapeComponentHandler(DataStore_ECS7 dataStoreEcs7)
         {
+            shapeRepresentation = new ShapeRepresentation();
```
Comment (pbosio) [inline unity-renderer/Assets/DCLPlugins/ECS7/ECSComponents/GLTFShape/Handler/GLTFShapeComponentHandler.cs:24]: ❤️

File: unity-renderer/Assets/DCLPlugins/ECS7/ProtocolBuffers/Definitions/Animator.proto [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+}
+
+message PBAnimationState {
+  string name = 1;
+  string clip = 2;
+  bool playing = 3;
+  float weight = 4;
+  float speed = 5;
+  bool loop = 6;
+  bool should_reset = 7;
```
Comment (pbosio) [inline unity-renderer/Assets/DCLPlugins/ECS7/ProtocolBuffers/Definitions/Animator.proto:17]: I think we should analyze if `should_reset` field applies to the data oriented components or if we should just rename it `reset`?
because, if I recall correctly, it was a hidden field set by the `Play()` method in previous ecs versions

**Review by pbosio — COMMENTED** [review]
I think we should seriously consider doing a proper refactor on this component

File: unity-renderer/Assets/DCLPlugins/ECS7/ECSComponents/Animator/Handler/AnimatorComponentHandler.cs (hunk 1 above)
Comment (Suduck) [reply unity-renderer/Assets/DCLPlugins/ECS7/ECSComponents/Animator/Handler/AnimatorComponentHandler.cs:71]: Fixed

File: unity-renderer/Assets/DCLPlugins/ECS7/ECSComponents/Animator/Handler/AnimatorComponentHandler.cs (hunk 2 above)
Comment (Suduck) [reply unity-renderer/Assets/DCLPlugins/ECS7/ECSComponents/Animator/Handler/AnimatorComponentHandler.cs:123]: Sure! If you think that there is a more efficient approach go ahead! I will add this to the issues card we have for that kind of optimization.
https://github.com/decentraland/sdk/issues/341

I will merge this since it is a working component right now and since we didn't release the new ECS7 yet, we can change this without problems

File: unity-renderer/Assets/DCLPlugins/ECS7/ProtocolBuffers/Definitions/Animator.proto (hunk 4 above)
Comment (Suduck) [reply unity-renderer/Assets/DCLPlugins/ECS7/ProtocolBuffers/Definitions/Animator.proto:17]: We can change that kind of name when we want. I think that we should take a look at all components names
